internal/api/dispatch/impl.go

```diff
@@ -106,3 +107,49 @@ func (this *dispatchManager) ProcessRun(ctx context.Context, account string, ser
 	instrumentation.RunCreated(ctx, run.Recipient, entity.ID, run.Url, entity.Service, protocol.GetLabel())
 	return entity.ID, correlationID, nil
 }
+
+func (this *dispatchManager) ProcessCancel(ctx context.Context, account string, cancel generic.CancelInput) (runID, correlationID uuid.UUID, err error) {
+	var run db.Run
+
+	if err := this.db.First(&run, cancel.RunId).Error; err != nil {
+		instrumentation.PlaybookRunCancelError(ctx, err)
+		return uuid.UUID{}, run.CorrelationID, &RunNotFoundError{err: err, runID: *cancel.RunId}
+	}
+
+	if run.SatId == nil || run.SatOrgId == nil {
+		instrumentation.PlaybookRunCancelRunTypeError(ctx, run.ID)
+		return uuid.UUID{}, run.CorrelationID, &RunCancelTypeError{err, run.ID}
+	}
+
+	if run.Status != "running" {
+		return uuid.UUID{}, run.CorrelationID, &RunCancelNotCancelableError{run.ID}
+	}
+
+	protocol := *protocols.SatelliteProtocol
+	signalMetadata := protocol.BuildCancelMetadata(cancel, run.CorrelationID, this.config)
+
+	// take from the rate limit bucket
+	this.rateLimiter.Wait(ctx)
+
+	messageId, notFound, err := this.cloudConnector.SendCloudConnectorRequest(
+		ctx,
+		account,
+		run.Recipient,
+		run.PlaybookRunUrl,
```

Comment:
oh, for cancel we should not be sending the URL. The SendCloudConnectorRequest method will need to be updated to make the url (content) optional